proj1a/LinkedListDeque.java

```diff
@@ -0,0 +1,145 @@
+public class LinkedListDeque<T> implements Deque<T> {
+    private class LLDNode {
+        LLDNode prev;
+        T item;
+        LLDNode next;
+
+        public LLDNode(LLDNode front, T n, LLDNode back) {
+            prev = front;
+            item = n;
+            next = back;
+        }
+    }
+
+    private LLDNode sentinel;
+    private int size;
+
+    /**create empty linked list deque*/
+    public LinkedListDeque() {
+        sentinel = new LLDNode (null, null, null);
+        sentinel.prev = sentinel;
+        sentinel.next = sentinel;
+        size = 0;
+    }
+
+    /**add item to front of the deque*/
+    @Override
+    public void addFirst(T item) {
+        LLDNode temp = new LLDNode(sentinel, item, sentinel.next);
+        sentinel.next.prev = temp;
+        sentinel.next = temp;
+        size = size + 1;
+    }
+
+    /**add item to back of the deque*/
+    @Override
+    public void addLast(T item) {
+        LLDNode temp = new LLDNode(sentinel.prev, item, sentinel);
+        sentinel.prev.next = temp;
+        sentinel.prev = temp;
+        size = size + 1;
+    }
+
+    /**return if deque is empty*/
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    /**number of items in the deque*/
+    @Override
+    public int size() {
+        return size;
+    }
+
+    /**print items in the deque*/
+    @Override
+    public void printDeque() {
+        LLDNode p = sentinel.next;
+        while (p != sentinel) {
+            System.out.print(p.item + " ");
+            p = p.next;
+        }
+        System.out.println();
+    }
+
+    /**remove and return item at front of deque*/
+    @Override
+    public T removeFirst() {
+        /**no first item*/
+        if (isEmpty()) {
+            return null;
+        }
+
+        T result = sentinel.next.item;
+        LLDNode target = sentinel.next.next;
+        LLDNode temp = new LLDNode(sentinel, target.item, target.next);
+        sentinel.next = temp;
```

Comment:
Don't quite understand the above code. Why adding another `LLDNode`? Shouldn't we remove the first `LLDNode` by making sure not reference to it? Would the number of `LLDNode` be inconsistent with `size`?